Drivers/Src/LTC2315.c

```diff
@@ -58,22 +58,29 @@ void LTC2315_Sleep() {
  */
 uint16_t LTC2315_Read() {
     OS_ERR err;
-    uint8_t rxdata[2];
+    uint8_t rxdata[2] = {0xff, 0xff};
+
+    LTC2315_wakeup_sleep();
 
     uint8_t count = 0;
     do {
-    LTC2315_wakeup_sleep();
+    OSSchedLock(&err);
+    assertOSError(err);
 
     BSP_SPI_SetStateCS(spi_ltc2315, 0);
     BSP_SPI_Read(spi_ltc2315, rxdata, 2);
```

Comment:
Should we add a note here that if we send more than 8 bytes, the OS will fault.

Reply:
I can add the comment